ui/round/src/keyboardMove.ts

```diff
@@ -28,6 +35,19 @@ export function ctrl(root: RoundController, step: Step, redraw: Redraw): Keyboar
   let usedSan = false;
   const cgState = root.chessground.state;
   return {
+    drop(key, piece) {
+      const role = (sanToRole as SanMap)[piece];
+      const crazyData = root.data.crazyhouse;
+      const color = root.data.player.color;
+      // Piece not in Pocket
+      if (!role || !crazyData || !crazyData.pockets[color === 'white' ? 0 : 1][role]) return;
+      // Square occupied
+      if (root.chessground.state.pieces[key]) return;
+      if (!crazyValid(root.data, role, key)) return;
+      root.chessground.cancelMove();
+      root.chessground.newPiece({ role, color }, key);
+      root.sendNewPiece(role, key, false);
```

Comment:
what happens if the drop is illegal (because of a check), is the new piece reverted?

Reply:
I don't think it passes validation, so it shouldn't get here in the first place.